test/markups/README.mdoc.html

```diff
@@ -0,0 +1,20 @@
+<html><body>
+<table style="width: 100%;">
+  <tr>
+    <td class="head-ltitle">README(1)</td>
+    <td style="text-align: center;">General Commands Manual</td>
+    <td style="text-align: right;">README(1)</td>
+  </tr>
+</table>
+<div class="manual-text">
+<h1 class="Sh" title="Sh" id="README.mdoc"><a class="selflink" href="#README.mdoc">README.mdoc</a></h1>
```

Comment:
This markup was generated by an older version of `mandoc`; specifically, one which preceded the recent improvements to its HTML output.

The most recent `mandoc` release is 1.14.4 – it's not easy to tell as the program lacks support for a `--version` switch.

@ischwarze Would you consider implementing a `-v | --version` switch in a future `mandoc` release? The current behaviour is far from helpful:
~~~console
λ ~: mandoc --version
mandoc: illegal option -- -
mandoc: illegal option -- v
mandoc: illegal option -- e
mandoc: illegal option -- r
usage: mandoc [-ac] [-I os=name] [-K encoding] [-mdoc | -man] [-O options]
              [-T output] [-W level] [file ...]
λ ~: mandoc -v
mandoc: illegal option -- v
usage: mandoc [-ac] [-I os=name] [-K encoding] [-mdoc | -man] [-O options]
              [-T output] [-W level] [file ...]
~~~

Reply:
@Alhadis Fixed - I generated the test artifact on my workstation, which is a Mac, and Homebrew currenly packages 1.14.3 so I ended up using that by accident.

Reply:
[So I see](https://github.com/Homebrew/homebrew-core/blob/bcc4e1032b479eb5fd303fe96a9f99cd03563d06/Formula/mandoc.rb#L4). I'll send them a pull-request to bump it to `1.14.4`. =)

Reply:
Ouch, i totally missed this question and only found it now.  No, there is no way i will add a --version option.  We used to have such an option years ago, and the result was a tremendous waste of time.  In bug reports, people used to mention nothing but that number, which told me absolutely nothing, so it was an endless back-and-forth with bug reporters to figure out what they were actually running.  Once i deleted the --version option, the situation got much better, people starting actually reporting what really matters, like the operating system name and OS version, whether they compiled from source, and if so, from which tarball or from a CVS checkout from which date, or which package manager they used and which version of the package.  You know, 1.14.4 from FreeBSD base, FreeBSD packages, NetBSD base, NetBSD packages = pkgsrc, Debian packages, homebrew, MacPorts, the raw release tarball, or checked out from CVS and so on and so forth are totally different things, they all contain different patches...

Reply:
Couldn't you include the information you're looking for (OS, commit number, etc.) in the output of the `--version` command? That way people would just copy it and you would be sure to get the information you need. You wouldn't be the first to proceed that way either...

Reply:
@pchaigno No - it should be obvious from what i said above that isn't possible.  Almost all downstream packagers apply local patches in their build systems - often for good reasons, occasionally due to misunderstandings - when building the mandoc packages that people then install with a wide variety of package management tools, so there is no such thing as a "commit number".  Also, the mandoc program itself cannot possibly know by which package manager it has been installed, and even the way to find out which operating system version you are running on is operating system dependent, so doing it with a program is very tricky and fragile, while it is trivial to report for a human user.  The whole concept of --version is totally broken, poorly thought out, and unfixable.

Reply:
Okay, first of all, a `--version` string isn't some magical keyword that'll unlock the hidden door to the underside of the program which Satan drilled when you weren't looking to [sneak this in](https://gist.github.com/aras-p/6224951).

I seriously doubt having a `--version` string supported yielded THAT many lazy reports (especially users of a system who know better than most to RTFM), and the virtue of providing helpful system diagnostics or distro-specific details should be a given for any halfway insightful hacker.

The `--version` string needn't be anything more insightful than "the point in development when this release was cut", which will help identify missing features when they get rolled out in future. If, for example, `mandoc` decides to support ditroff or JSON-encoded serialisations of its output for consumption by other tools, being able to check a switch to determine if the version supports it is one of the reasons things like this exist.

My tirade notwithstanding, it should be pretty obvious that a `--version` switch has value beyond the most obvious in bug reports.

Reply:
Of course, you don't have to add the `--version` switch if you prefer not to. 😉 I'm already tempted to jokingly write up a Roff script which detects mandoc's precise version based on how many low-level Roff syntax it doesn't support, and compare those with the most recent improvements `mandoc` has made concerning general-purpose typesetting. ;)

Reply:
Yeah, but that's Perl, and we all know the Perl folk can't do any wrong when it comes to anything text-related. 😉 (That includes being Ruby's superior at text-processing `thereisaidit` #HAIL_TEAM_PERL#.

Jests and half-jests aside, most programs needn't have a very elaborate `--version` output, especially if they're small and too uncomplicated to do so anyway...

Reply:
Sounds like a version option isn't something @ischwarze is interested in adding, so let's leave this thread here unless there's something new to say.